bundle/edu.gemini.qpt.client/src/main/java/edu/gemini/qpt/core/Marker.java

```diff
@@ -24,9 +21,15 @@
     private final String text;
     private final Severity severity;
     private final boolean qcOnly;
-    private final Option<Union<Interval>> union;
-
-    public Marker(boolean qcOnly, Object owner, Severity severity, String text, Option<Union<Interval>> union, Object... path) {
+    private final Option<Pair<ImList<String>, Union<Interval>>> union;
+
+    public Marker(boolean qcOnly,
+                  Object owner,
+                  Severity severity,
+                  String text,
+                  Option<Pair<ImList<String>, Union<Interval>>> union,
+                  Object... path
+    ) {
```

Comment:
Getting the timestamps formatted in the message is tricky.  I was thinking that the `String text` argument would potentially be a format string itself with made up placeholder variables for each timestamp that we need to format.  Then the next argument would be a `List<Long>` or `ImList<Long>` of matching timestamps to format.  Nothing would guarantee that the variables in `text` would be the same number as the elements in the list of course, but I believe it would be clearer what is going to happen.

So for example, the `text` might be `"Must be observed between %t(A) - %t(sunrise)."` and we'd expect two corresponding `Long` arguments in the given `ImList`.  So basically fold the format `text` over the `List<Long>` replacing the first `%t` if any with the current `Long`, formatted as `"HH:mm"` in the right timezone of course, at each step.